### PR TITLE
Init base pose from datastore

### PR DESCRIPTION
**Pull request by orikuma**
When I start MCC after finishing another controller, the initial pose of the robot can be different from the one assumed in pre-defined motion in MCC. In addition, when I change a controller from MCC to other controllers and go back to MCC, the pose of the base can be modified by other controllers though the actual robot is in the same position.
To solve this initialization problem, I add option to save last pose of the base in datastore and overwrite pose of the base link in reset function. It also can be defined from yaml file as basePose, which is loaded when instantiating MCC.

### CODE REVIEW

File: include/MultiContactController/states/ConfigMotionState.h [hunk 1]
```diff
@@ -25,5 +25,11 @@ struct ConfigMotionState : State
 
   //! Task configuration list
   std::multimap<double, mc_rtc::Configuration> taskConfigList_;
+
+  //! Option to select whether this state should wait for finishing swing motion or not
+  bool exitWhenLimbSwingFinished_ = false;
+
+  //! Option to save the last pose in datastore for the other motion
+  bool saveLastBasePose_ = false;
```
Comment (mmurooka) [inline include/MultiContactController/states/ConfigMotionState.h:33]: This variable defined in ConfigMotionState seems to be unused.

File: src/states/ConfigMotionState.cpp [hunk 2]
```diff
@@ -89,6 +89,16 @@ void ConfigMotionState::start(mc_control::fsm::Controller & _ctl)
     }
   }
 
+  // Set option to wait for finishing swing motion
+  if(config_.has("configs") && config_("configs").has("exitWhenLimbSwingFinished"))
+  {
+    exitWhenLimbSwingFinished_ = static_cast<bool>(config_("configs")("exitWhenLimbSwingFinished"));
+  }
+  if(config_.has("configs") && config_("configs").has("saveLastBasePose"))
```
Comment (mmurooka) [inline src/states/ConfigMotionState.cpp:97]: This variable defined in ConfigMotionState seems to be unused.

File: include/MultiContactController/MultiContactController.h [hunk 3]
```diff
@@ -89,6 +89,9 @@ struct MultiContactController : public mc_control::fsm::Controller
   //! Whether to enable manager update
   bool enableManagerUpdate_ = false;
 
+  //! Whether to save last base pose when stopping this controller
```
Comment (mmurooka) [inline include/MultiContactController/MultiContactController.h:92]: Please comment here that the `basePose` configuration allows you to set the initial value and the `saveLastBasePose` configuration allows you to save the base pose (if you have a better place to do it, that's fine too).

**Review by mmurooka — APPROVED** [review]
Thank you for the PR! Some small comments.

I believe this feature should be included in the mc_rtc controller definition itself, as it will be needed not only for MCC, but you can merge it for quick development for now.

File: include/MultiContactController/states/ConfigMotionState.h (hunk 1 above)
Comment (orikuma) [reply include/MultiContactController/states/ConfigMotionState.h:33]: Fixed in da202a418afb2cad57c4d6eb1b87e0eda5febd5c

File: src/states/ConfigMotionState.cpp (hunk 2 above)
Comment (orikuma) [reply src/states/ConfigMotionState.cpp:97]: Fixed in da202a418afb2cad57c4d6eb1b87e0eda5febd5c

File: include/MultiContactController/MultiContactController.h (hunk 3 above)
Comment (orikuma) [reply include/MultiContactController/MultiContactController.h:92]: I added the comment for constructor, reset, and stop functions in da202a418afb2cad57c4d6eb1b87e0eda5febd5c

**Pull request closed without merging**